packages/grid/_modules_/grid/hooks/features/useColumnResize.tsx

```diff
@@ -1,204 +1,131 @@
-      columnsRef.current.addEventListener('mouseup', stopResize);
-      columnsRef.current.addEventListener('mouseleave', handleMouseLeave);
-      columnsRef.current.addEventListener('mouseenter', handleMouseEnter);
-      columnsRef.current.addEventListener('mousemove', handleMouseMove);
-
-      return () => {
-        columnsRefEvents.removeEventListener('mouseup', stopResize);
-        columnsRefEvents.removeEventListener('mouseleave', handleMouseLeave);
-        columnsRefEvents.removeEventListener('mouseenter', handleMouseEnter);
-        columnsRefEvents.removeEventListener('mousemove', handleMouseMove);
-      };
+    // Skip if the column isn't resizable
+    if (!event.currentTarget.classList.contains('MuiDataGrid-columnSeparatorResizable')) {
+      return;
     }
-    return undefined;
-  }, [
-    columnsRefState,
-    columnsRef,
-    handleMouseLeave,
-    handleMouseDown,
-    handleMouseMove,
-    handleMouseEnter,
-    logger,
-    stopResize,
-  ]);
+
+    // Avoid text selection
+    event.preventDefault();
+
+    colElementRef.current = event.currentTarget.closest(
+      `.${HEADER_CELL_CSS_CLASS}`,
+    ) as HTMLDivElement;
+    const field = colElementRef.current.getAttribute('data-field') as string;
+    const colDef = apiRef.current.getColumnFromField(field);
+
+    logger.debug(`Start Resize on col ${colDef.field}`);
+    apiRef.current.publishEvent(COL_RESIZE_START, { field });
+
+    colDefRef.current = colDef;
+    colElementRef.current = columnsRef.current!.querySelector(
+      `[data-field="${colDef.field}"]`,
+    ) as HTMLDivElement;
+
+    colCellElementsRef.current = findCellElementsFromCol(colElementRef.current) as NodeListOf<
+      Element
+    >;
+
+    const doc = ownerDocument(apiRef.current.rootElementRef!.current as HTMLElement);
+    doc.body.style.cursor = 'col-resize';
+
+    initialOffset.current =
+      (colDefRef.current.width as number) -
+      (event.clientX - colElementRef.current!.getBoundingClientRect().left);
+
+    // @ts-expect-error fixed in v5
+    doc.addEventListener('mousemove', handleResizeMouseMove);
+    // @ts-expect-error fixed in v5
+    doc.addEventListener('mouseup', handleResizeMouseUp);
+  });
+
+  const stopListening = React.useCallback(() => {
+    const doc = ownerDocument(apiRef.current.rootElementRef!.current as HTMLElement);
+    doc.body.style.removeProperty('cursor');
+    // @ts-expect-error fixed in v5
+    doc.removeEventListener('mousemove', handleResizeMouseMove);
+    // @ts-expect-error fixed in v5
+    doc.removeEventListener('mouseup', handleResizeMouseUp);
+  }, [apiRef, handleResizeMouseMove, handleResizeMouseUp]);
 
   React.useEffect(() => {
     return () => {
-      clearTimeout(mouseLeftTimeout.current);
       clearTimeout(stopResizeEventTimeout.current);
+      stopListening();
     };
-  }, []);
+  }, [stopListening]);
 
-  return handleMouseDown;
+  return React.useMemo(() => ({ onMouseDown: handleMouseDown }), [handleMouseDown]) as {};
```

Comment:
why the as {} and not {onMouseDown:()=>...}   ?

Reply:
Updated to `React.HTMLAttributes<HTMLDivElement>`.